src/processors/tone/ladder_filter/LadderFilterProcessor.cpp

```diff
@@ -0,0 +1,135 @@
+#include "../../ParameterHelpers.h"
+
+#include "LadderFilterProcessor.h"
+
+LadderFilterProcessor::LadderFilterProcessor (UndoManager* um) : BaseProcessor ("Ladder Filter", LadderParameters::createParameterLayout(), um), p (vts)
+{
+    // Info and style
+    uiOptions.info.description = "Resonant 4-pole ladder high-pass and low-pass,\n capable of self-oscillation. With nonlinear drive.";
+    uiOptions.info.authors = StringArray { "Butch Warns" };
+    uiOptions.info.infoLink = "https://butchwarns.de/";
+    uiOptions.backgroundColour = juce::Colour::fromRGB (0xBB, 0xE1, 0xC3);
+    uiOptions.powerColour = juce::Colour::fromRGB (0x99, 0x0B, 0x3C);
+
+    addPopupMenuParameter ("FILTER_MODE");
+}
+
+void LadderFilterProcessor::prepare (double sampleRate, int samplesPerBlock)
+{
+    typedef LadderParameters params;
+    drive_smooth.mappingFunction = params::drive;
+    hp_cutoff_smooth.mappingFunction = params::hp_cutoff;
+    hp_resonance_smooth.mappingFunction = params::hp_resonance;
+    lp_cutoff_smooth.mappingFunction = params::lp_cutoff;
+    lp_resonance_smooth.mappingFunction = params::lp_resonance;
+
+    drive_smooth.setParameterHandle (p.drive_norm);
+    drive_norm_smooth.setParameterHandle (p.drive_norm);
+    hp_cutoff_smooth.setParameterHandle (p.hp_cutoff_norm);
+    hp_resonance_smooth.setParameterHandle (p.hp_resonance_norm);
+    lp_cutoff_smooth.setParameterHandle (p.lp_cutoff_norm);
+    lp_resonance_smooth.setParameterHandle (p.lp_resonance_norm);
+
+    drive_smooth.prepare (sampleRate, samplesPerBlock);
+    drive_norm_smooth.prepare (sampleRate, samplesPerBlock);
+    hp_cutoff_smooth.prepare (sampleRate, samplesPerBlock);
+    hp_resonance_smooth.prepare (sampleRate, samplesPerBlock);
+    lp_cutoff_smooth.prepare (sampleRate, samplesPerBlock);
+    lp_resonance_smooth.prepare (sampleRate, samplesPerBlock);
+
+    // Initialize filters
+    for (int channel = 0; channel < 2; ++channel)
+    {
+        lp[channel].reset (sampleRate);
+        hp[channel].reset (sampleRate);
+    }
+}
+
+void LadderFilterProcessor::processAudio (AudioBuffer<float>& buffer)
+{
+    const int num_samples = buffer.getNumSamples();
+
+    for (int channel = 0; channel < buffer.getNumChannels(); ++channel)
+    {
+        float* data = buffer.getWritePointer (channel);
+
+        // Process high-pass
+
+        // If we need smoothing..
+        hp_cutoff_smooth.process (num_samples);
```

Comment:
So the `SmoothedBufferValue::process()` calls should happen _outside_ the channels loop, since otherwise we'd be running the smoother on the same parameter twice per block.

Reply:
Absolutely right, done.